src/Features/Core/Portable/ReplacePropertyWithMethods/ReplacePropertyWithMethodsCodeRefactoringProvider.cs

```diff
@@ -0,0 +1,414 @@
+                string.Format(resourceString, propertyName),
+                c => ReplacePropertyWithMethods(context.Document, propertySymbol, c),
+                propertyName));
+        }
+
+        private async Task<Solution> ReplacePropertyWithMethods(
+           Document document,
+           IPropertySymbol propertySymbol,
+           CancellationToken cancellationToken)
+        {
+            var desiredMethodSuffix = NameGenerator.GenerateUniqueName(propertySymbol.Name,
+                n => !HasAnyMatchingGetOrSetMethods(propertySymbol, n));
+
+            var desiredGetMethodName = GetPrefix + desiredMethodSuffix;
+            var desiredSetMethodName = SetPrefix + desiredMethodSuffix;
+
+            var semanticModel = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+
+            var originalSolution = document.Project.Solution;
+            var propertyReferences = await SymbolFinder.FindReferencesAsync(propertySymbol, originalSolution, cancellationToken).ConfigureAwait(false);
+
+            // Get the warnings we'd like to put at the definition site.
+            var definitionWarning = GetDefinitionIssues(propertyReferences);
+
+            var equalityComparer = (IEqualityComparer<IPropertySymbol>)SymbolEquivalenceComparer.Instance;
+            var definitionToBackingField = 
+                propertyReferences.Select(r => r.Definition)
+                                  .OfType<IPropertySymbol>()
+                                  .ToDictionary(d => d, GetBackingField, equalityComparer);
+
+            var q = from r in propertyReferences
+                    where r.Definition is IPropertySymbol
+                    from loc in r.Locations
+                    select ValueTuple.Create((IPropertySymbol)r.Definition, loc);
+
+            var referencesByDocument = q.ToLookup(t => t.Item2.Document);
+
+            // References and definitions can overlap (for example, references to one property
+            // inside the definition of another).  So we do a multi phase rewrite.  We first
+            // rewrite all references to point at the new methods instead.  Then we remove all
+            // the actual property definitions and replace them with the new methods.
+            var updatedSolution = originalSolution;
+
+            updatedSolution = await UpdateReferencesAsync(
+                updatedSolution, referencesByDocument, definitionToBackingField, 
+                desiredGetMethodName, desiredSetMethodName, cancellationToken).ConfigureAwait(false);
+
+            updatedSolution = await ReplaceDefinitionsWithMethodsAsync(
+                originalSolution, updatedSolution, propertyReferences, definitionToBackingField, 
+                desiredGetMethodName, desiredSetMethodName, cancellationToken).ConfigureAwait(false);
+
+            return updatedSolution;
+        }
+
+        private bool HasAnyMatchingGetOrSetMethods(IPropertySymbol property, string name)
```

Comment:
Ah yes.  This was crashing during editing (because of the lambda bug).  So i didn't get around to actually implementing it :)